packages/kit/src/core/config/options.js

```diff
@@ -108,7 +109,11 @@ const options = object(
 			}),
 
 			csrf: object({
-				checkOrigin: boolean(true),
+				checkOrigin: deprecate(
+					boolean(true),
+					(keypath) =>
+						`\`${keypath}\` has been deprecated in favour of \`csrf.trustedOrigins: ['*']\`. It will be removed in a future version`
```

Comment:
Maybe just listing the option name would be better, requiring the developer to lookup the new option instead?
```suggestion
						`\`${keypath}\` has been deprecated in favour of \`csrf.trustedOrigins\`. It will be removed in a future version`
```

> what would be the correct value for "only allow deployed origin", empty array or not setting it?

I'm guessing the deployed origin is always allowed so both and empty array and undefined would be equivalent, but I'm not sure.

Reply:
yeah, omitting the value seems reasonable